README.md

```diff
@@ -1,125 +1,206 @@
+As ECDSA cryptographic signature following the EIP-712 standard which we use to verify the authenticity of an `Order`. You will see the `Components` of a Signature (V, R, and S) used throughout the protocol. The Hash and Sig solidity libraries exist to service this functionality. It should be noted and enforced that `Components` are derived from a `Signature` and that the two are not equivalent. a `Signature` is a 64-bytes and its derived `Components` are a _tuple_ of length 3 consisting of two 32-bytes, the canonical _R_ and _S_,  and a single uint _V_.
 
-### Transfer
-Terms such as transfer, transfers  and transfer from when referencing specific functionality of the Underlying token.
-As with approve, specificy which token if necessary, _uToken transfer_ etc...
+An Order's signature is considered `valid` if the extracted public key matches that of the maker.
+### EIP-712 Domain
+Mentioned here as the term `Domain` has very specific Domain Driven Design implications. You may see or hear the term domain intermingled with other verbage for the EIP-712 standard such as:
+* TypeHash
+* Separator
+* Message
 
-### Balance
-Amount of the underlying token a given public key owns. Specify which token to avoid confusion.
+Any use of these terms, specifically _domain_ should be prefixed with EIP-712 to prevent language degradation and the confusion that follows it.
 
-## Compound Token
-A CErc20 at an address passed to the constructor of the Swivel contract at deployement.
-* Perhaps pass an array of underlying token addresses + array of CERC20 addresses to create a public mapping of address -> address 
+Implementation of Signature and Components are each tightly coupled to the EIP-712 standard.
+### Initiate Event
+Emitted on chain upon the establishment of any `Agreement`. Publishes both `OrderKey` and `AgreementKey`
+### Release Event
+Emitted on chain upon the release of an Agreement. Publishes OrderKey and AgreementKey
+### Approve
+Terms such as approve and approval(s) when referencing specific functionality of the `Underlying` token. Use specifiers to avoid confusion with the compound token if needed. _underlying approval_ for example.
+### Transfer
+Terms such as transfer, transfers  and transfer from when referencing specific functionality of the `Underlying` token. As with approve, specify which token if necessary, _uToken transfer_ etc...
+### Side
+Indicates whether the bond is `fixed` and is lending `principal`, or is `floating` and is paying `interest`.
+### Exit
+Boolean value indicating that an `Order` is Selling `nTokens` of `zcTokens` back to the `Orderbook`.
+### Vault
+An on-chain contract that tracks the token balances for a given market for a given user.
+### Event Store
+The saved stream of order event data 
+### Event Sourcing
+The data strategy for storing and retrieving events from the event store.
+### Messages
+Messages are published with predefined meanings that are used to observe the occurance of events.
+### Initiate
```

Comment:
wut? is this a V1 leftover? seems like it is. `Initiate` now is opposed to `Exit` - pretty sure it's explained elsewhere